frontend/src/pages/eventoverview.js

```diff
@@ -0,0 +1,174 @@
+import React from 'react'
+import Router from 'next/router'
+import {
+  Table,
+  Row,
+  Button,
+  Col,
+  Form,
+  FormGroup,
+  Label,
+  Input,
+  Container,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter
+} from 'reactstrap'
+import ActionButton from '../components/actionButton'
+import Nav from '../components/nav'
+import { getAllEvents, createEvent } from '../utils/api'
+
+class EventOverview extends React.Component<Props> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      events: [],
+      modal: false,
+      name: '',
+      date: '',
+      startTime: 0,
+      endTime: 0,
+      location: '',
+      description: ''
+    }
+  }
+
+  async componentDidMount() {
+    const events = await getAllEvents()
+    if (events) {
+      this.setState({
+        events: events.result
+      })
+    }
+  }
+
+  openModal = () => {
+    this.setState({ modal: true })
+  }
+
+  closeModal = () => {
+    this.setState({ modal: false })
+  }
+
+  handleChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    })
+  }
+
+  handleKeyPress = e => {
+    if (e.charCode === 13) {
+      this.addEvent()
+    }
+  }
+
+  addEvent = async () => {
+    let event = {
+      name: this.state.name,
+      date: this.state.date,
+      startTime: this.state.startTime,
+      endTime: this.state.endTime,
+      location: this.state.location,
+      description: this.state.description
+    }
+    await createEvent(event)
```

Comment:
Are there any checks anywhere to make sure that these values have been set? These values are required in the backend but if I leave it empty it will (should) throw an error